### DISCUSSION TITLE
Search / filter by log context

### DISCUSSION THREAD

**Discussion by PHLAK** [opcodesio/log-viewer#286] [Ideas]
I would like to be able search / filter logs by context. For example, given the following log call in my app.

```php
Log::info('Some log message', ['some_identifier' => '12345']);
```

It would be nice if I could search for `some_identifier:12345` (or something similar) and see only log messages with the context of `'some_identifier' => '12345'`.

This is something I've been able to set up with [Tagging in Laravel Telescope](https://laravel.com/docs/10.x/telescope#tagging) but the Telescope log viewer isn't quite as suitable to viewing lots of logs. I much prefer this package for that.

Right now this is indirectly possible by searching for `"some_identifier":"12345"` since it searches the _raw_ log context. However, this isn't very intuitive or immediately apparent since log contexts are pretty printed (i.e. with spaces and newlines added for formatting) causing the output and search string to not match.

**Comment by arukompas** [comment]
That would be a great feature of course, but is more complex to develop :) I might have some time later this year, but if you feel like you can also tackle this, feel free to contribute a PR!

**Reply by PHLAK** [reply]
> That would be a great feature of course, but is more complex to develop

I totally understand. Just wanted to make the suggestion.

> I might have some time later this year, but if you feel like you can also tackle this

Unfortunately I don't have a lot of spare time right now either. However, if that changes I might look into it.
